### PR TITLE
Add missing fixes from OpenBSD ports tree

### PR DESCRIPTION
**Pull request by czarkoff**
Apparently most patches from OpenBSD ports tree already made their way into i3. These remain.

### CODE REVIEW

**Comment by stapelberg** [comment]
Can you please use clang-format to format the code to make travis happy?

**Comment by czarkoff** [comment]
Now it builds for me with no patchwork both on OpenBSD and Linux, and travis is happy.

**Comment by czarkoff** [comment]
If you consider merging this pull request without collapsing commits, please notify me - I'd rather purge *Home Server* from history and rearrange diffs on my side.

File: Makefile [hunk 1]
```diff
@@ -72,6 +68,11 @@ CFLAGS += -idirafter yajl-fallback
 OBJS:=$(wildcard src/*.c *.c)
 OBJS:=$(OBJS:.c=.o)
 
+ifeq ($(OS),OpenBSD)
+OBJS:=$(filter-out src/pulse.o, $(OBJS))
+LIBS:=$(filter-out -lpulse, $(LIBS)) -lpthread
```
Comment (stapelberg) [inline Makefile]: Why are you compiling without PulseAudio on OpenBSD? PulseAudio is available and actively maintained (AFAICT) on OpenBSD: http://openports.se/audio/pulseaudio

**Comment by czarkoff** [comment]
Sure, but it plays completely different role. On Linux PulseAudio manages audio devices. On OpenBSD it is just an interface on top of SndIO, and most programs speak to SndIO directly. I am not even sure that PulseAudio volume settings actually affect the system-wide volume on OpenBSD.

FWIW you might notice that most OpenBSD ports are compiled without pulseaudio support if it is optional.

File: Makefile (hunk 1 above)
Comment (czarkoff) [inline Makefile]: Sure, but it plays completely different role. On Linux PulseAudio manages audio devices. On OpenBSD it is just an interface on top of SndIO, and most programs speak to SndIO directly. I am not even sure that PulseAudio volume settings actually affect the system-wide volume on OpenBSD.

FWIW you might notice that most OpenBSD ports are compiled without pulseaudio support if it is optional.

**Comment by stapelberg** [comment]
Can you squash your commits please?

**Pull request closed without merging**